### ISSUE TITLE
Schedule notification that repeats every 4 days at a given time

### ISSUE THREAD

**Issue by Nek-12** [Tweener/alarmee#1]
How can I schedule notifications repeating at regular intervals?

**Comment by Tweener** [comment]
> How can I schedule notifications repeating at regular intervals?

Hi @Nek-12, at the moment it is not possible to schedule repeated alarms. But this is in the roadmap for this library so stay tuned for updates!

**Comment by Tweener** [comment]
Hey @Nek-12 , with version 1.2.0 of this library, you can now schedule repeated alarms by providing a `RepeatInterval`.
Details [here](https://github.com/Tweener/alarmee?tab=readme-ov-file#scheduling-an-alarm).

Let me know if this is what you need!

**Comment by Nek-12** [comment]
No, this doesn't solve the problem. I want to repeat the notification every 4 days, but with the provided options, I can only choose 1 day, 1 month, 1 hour or 1 year as an interval. 

Additionally, it looks like the notification doesn't come correctly when I set both first occurrence date (e.g. 24th of Nov 2024 and the repeat interval)

**Comment by Nek-12** [comment]
That option doesn't seem to work correctly. I still can't schedule a notification that **starts** at 6pm tomorrow, **then** repeats at **that time (6pm)** every 4 days. Instead, it just comes at 6pm tomorrow, but then repeats at the time it was scheduled, for example 11:24:55 am, not at the time I wanted (6pm).

**Comment by Tweener** [comment]
Yes, you can achieve this behavior by setting `scheduledDateTime` to tomorrow at 6pm and using `RepeatInterval.Custom(duration = 4.days)` like this:
```Kotlin
alarmeeScheduler.schedule(
    alarmee = Alarmee(
        uuid = "myAlarmId",
        notificationTitle = "🔁 Congratulations! You've schedule a custom repeating Alarmee!",
        notificationBody = "This notification will be displayed on December 18th, 2024, at 6:00 PM and will repeat every 4 days at 6:00 PM",
        scheduledDateTime = LocalDateTime(year = 2024, month = Month.DECEMBER, dayOfMonth = 18, hour = 18, minute = 0),
        repeatInterval = RepeatInterval.Custom(duration = 4.days), // Will repeat every 4 days
        androidNotificationConfiguration = AndroidNotificationConfiguration( // Required configuration for Android target only (this parameter is ignored on iOS)
            priority = AndroidNotificationPriority.DEFAULT,
            channelId = "otherChannelId",
        )
    )
)
```

The first notification will be displayed tomorrow at 6pm.
The second notification will be displayed exactly 4 days after tomorrow at 6pm, so the 22nd at 6pm. (4 days = 4*24*60*60*1000 = 345600000 milliseconds).
Etc.

Does that make sense?

**Comment by Nek-12** [comment]
Like I said, I tried it, but it doesn't work this way.

**Comment by Nek-12** [comment]
It seems that you're using UNTimeIntervalNotificationTrigger, which schedules the alarm to repeat **after** provided interval since scheduled, not the fixed time initially desired. https://developer.apple.com/documentation/usernotifications/untimeintervalnotificationtrigger?language=objc

**Comment by Tweener** [comment]
> Like I said, I tried it, but it doesn't work this way.

Can you show me the code you used to test this behavior? Because for sure you couldn't test with 4 days interval.

> To repeat, if I schedule my alarm right now, the first invocation should be at 4 pm on Wednesday, then the second one must be 4pm Sunday, not at 12:23pm Saturday.

That is not the expected behavior. The `scheduledDateTime` property defines the first occurence of the repeating alarm. So if you schedule your alarm right now (via `scheduledDateTime`), then the first notification will be triggered right now.

**Comment by Nek-12** [comment]
No, I scheduled it for the first occurrence date (in a day), but the repeat was 5 minutes, and as I thought, the notification came after 5 mins, not 20 hours as I specified.

Sorry, I don't have code, because to give you feedback quickly I threw together a test alarm and removed it afterwards locally on a work project

**Comment by Nek-12** [comment]
But afaik, there is no way to execute code upon the push notification being sent on iOS, so we can't reschedule the notification. I was hoping this library would be the solution to the problem you mentioned above, but if you think that's out of scope of the library, then this issue can be closed.
